backend/compact-connect/docs/design/README.md

```diff
@@ -31,6 +57,82 @@ efficient processing. A lambda receives messages from the SQS queue. Each messag
 ingested. The lambda receives the data and creates or updates a corresponding license record in the DynamoDB license
 data table.
 
-
 ### Asynchronous validation feedback
 Asynchronous validation feedback for boards to review is not yet implemented.
+
+## User Architecture
+[Back to top](#backend-design)
+
+Authentication with the CompactConnect backend will be controlled through Oauth2 via
+[AWS Cognito User Pools](https://github.com/csg-org/CompactConnect). Clients will be divided into two groups, each
+represented by an independent User Pool: [Staff Users](#staff-users) and [Licensee Users](#licensee-users). See
+the accompanying [architecture diagram](./users-arch-diagram.pdf) for an illustration.
+
+### Staff Users
+
+Staff users come with a variety of different permissions, depending on their role. There are Compact Executive
+Directors, Compact ED Staff, Board Executive Directors, Board ED Staff, and CSG Admins, each with different levels
+of ability to read and write data, and to administrate users. Read permissions are granted to a user for an entire
+compact or not at all. Data writing and user administration permissions can each be granted to a user per
+compact/jurisdiction combination. All of a compact user's permissions are stored in a DynamoDB record that is associated
+with their own Cognito user id. That record will be used to generate scopes in the Oauth2 token issued to them on login.
+See [Implementation of scopes](#implementation-of-scopes) for a detailed explanation of the design for exactly how
+permissions will be represented by scopes in an access token. See
+[Implementation of permissions](#implementation-of-permissions) for a detailed explanation of the design for exactly
+how permissions are stored and translated into scopes.
+
+#### Compact Executive Directors and Staff
+
+Compact ED level staff can have permission to read all compact data as well as to create and manage users and their
+permissions. They can grant other users the ability to write data for a particular jurisdiction and to create more
```

Comment:
It sounds like the ED level staff will have read and admin permissions.

But as an admin, one can create a user with write permissions, I think.

In that case, it seems like admin and write are implicitly intertwined, no? (If you have admin permissions, just create an account with write permissions.)

Reply:
And to clarify: What are those with "write" permissions able to write?

Reply:
> And to clarify: What are those with "write" permissions able to write?

They can write license/privilege data for their own jurisdiction.

> In that case, it seems like admin and write are implicitly intertwined, no?

Yes. There is a path to a sort of privilege escalation by virtue of a compact ED creating a board user with write permission for the board's jurisdiction. My understanding of the use cases here is that this is desirable. Compact EDs do not have any need or desire to write data for a jurisdiction themselves, but we aren't specifically aiming to prevent them from being able to do so, if they're determined to. Compact EDs _do_, however, need the ability to delegate write permissions to users in all jurisdictions for their compact, by virtue of creating board ED users.

Reply:
Got it, thanks!

Still a little confused on 'write' access. Does 'write' provide permission to add a new license? Edit existing licenses? If edit, which licenses can they edit? All of them? Is the assumption that if I have "write" access on a compact/jurisdiction, then I also have read access on the compact?

Reply:
So 'write' and 'read' permissions are entirely independent, the way this is set up. If you have 'read' access, you can read all license/privilege data for the whole compact. 'write' permission is specific to a jurisdiction. If a user has 'write' permissions for the Alabama jurisdiction for the Audiology/Speech Language Pathology compact, they will be able to create/update licenses, update privileges, but only within the Alabama jurisdiction, for ASLP.